### PR TITLE
Ci partial rework

### PR DESCRIPTION
**Pull request by carlopi**
Commits are independent.

### CODE REVIEW

File: .github/workflows/CodeQuality.yml [hunk 1]
```diff
@@ -14,7 +14,6 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  GH_TOKEN: ${{ secrets.GH_TOKEN }}
   TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
   AWS_ACCESS_KEY_ID: ${{ secrets.S3_ID }}
   AWS_SECRET_ACCESS_KEY: ${{ secrets.NODE_PRE_GYP_SECRETACCESSKEY }}
```
Comment (samansmink) [inline .github/workflows/CodeQuality.yml:19]: hey @carlopi looks good! i think one thing we also really want here is to move all of the secrets down to steps where we use them. We can have a look at this together tomorrow?

Comment (carlopi) [reply .github/workflows/CodeQuality.yml:19]: sure!
here it was mostly a trial-run on one of them, to check whether this all makes sense. Will go over the other ones then, and tomorrow we fill the gaps.

File: .github/workflows/Java.yml [hunk 2]
```diff
@@ -53,6 +51,9 @@ jobs:
           (cd examples/jdbc; make; make maven)
       - name: Deploy
         shell: bash
+        if: github.ref == 'refs/heads/master'
```
Comment (Mause) [inline .github/workflows/Java.yml:54]: What's the rationale behind no longer publishing artifacts on branch builds?

Comment (carlopi) [reply .github/workflows/Java.yml:54]: Thanks for catching this, it's just wrong, the logic should be "only for tags", and it's wrong in a similar way in a bunch of other places.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**